### PR TITLE
SWARM-1107: Change the type of SecuredArchivePreparer to List<Map<Sting, Object>> for non main() way

### PR DESCRIPTION
**Pull request by emag**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
 - https://issues.jboss.org/browse/SWARM-1107
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

Motivation
----------
Thanks to SWARM-1047, you can get the data in a full list/tree shape
akin to the YAML itself. Now List of Map in yml is injected as List<Map>.

Thus, org.wildfly.swarm.keycloak.runtime.SecuredArchivePreparer.securityConstraints
should be List<Map<String, Object>> instead of List<String> and then it enables
to setup the Security Constraints without main().

Modifications
-------------
* Change the type of SecuredArchivePreparer.securityConstraints to List<Map<String, Object>>
* Remove unnecessary Parser for converting Map in yaml to String

Result
------
You can setup Keycloak Security Constraints without main().

### CODE REVIEW

**Review by kenfinnigan — APPROVED** [review]

**Comment by kenfinnigan** [comment]
add to whitelist

**Comment by kenfinnigan** [comment]
add to whitelist

**Comment by kenfinnigan** [comment]
retest this please

**Comment by kenfinnigan** [comment]
retest this please

File: fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java [hunk 1]
```diff
@@ -64,12 +66,15 @@ public void prepareArchive(Archive<?> archive) {
...
-                    .withRole(sc.roles().toArray(new String[sc.roles().size()]));
+        securityConstraints.forEach(sc -> {
+            SecurityConstraint securityConstraint = archive.as(Secured.class)
+                    .protect((String) sc.getOrDefault("url-pattern", "/*"));
+
+            ((List<String>) sc.getOrDefault("methods", Collections.emptyList()))
+                    .forEach(m -> securityConstraint.withMethod(m));
+
+            ((List<String>) sc.getOrDefault("roles", Collections.emptyList()))
+                    .forEach(r -> securityConstraint.withRole(r));
```
Comment (gastaldi) [inline fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java]: Could use `forEach(securityConstraint::withRole)`

File: fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java [hunk 2]
```diff
@@ -64,12 +66,15 @@ public void prepareArchive(Archive<?> archive) {
...
-            archive.as(Secured.class)
-                    .protect(sc.urlPattern())
-                    .withMethod(sc.methods().toArray(new String[sc.methods().size()]))
-                    .withRole(sc.roles().toArray(new String[sc.roles().size()]));
+        securityConstraints.forEach(sc -> {
+            SecurityConstraint securityConstraint = archive.as(Secured.class)
+                    .protect((String) sc.getOrDefault("url-pattern", "/*"));
+
+            ((List<String>) sc.getOrDefault("methods", Collections.emptyList()))
+                    .forEach(m -> securityConstraint.withMethod(m));
```
Comment (gastaldi) [inline fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java]: Could use forEach(securityConstraint::withMethod)

File: fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java [hunk 3]
```diff
@@ -64,12 +66,15 @@ public void prepareArchive(Archive<?> archive) {
...
         }
 
-        securityConstraints.forEach(scAsString -> {
-            SecurityConstraint sc = SecurityConstraintParser.parse(scAsString);
-            archive.as(Secured.class)
-                    .protect(sc.urlPattern())
-                    .withMethod(sc.methods().toArray(new String[sc.methods().size()]))
-                    .withRole(sc.roles().toArray(new String[sc.roles().size()]));
+        securityConstraints.forEach(sc -> {
+            SecurityConstraint securityConstraint = archive.as(Secured.class)
```
Comment (gastaldi) [inline fractions/keycloak/src/main/java/org/wildfly/swarm/keycloak/runtime/SecuredArchivePreparer.java]: can this call be moved out of the forEach loop?

**Review by gastaldi — APPROVED** [review]
Added some comments, but nothing that would block merging this PR. Good job @emag !

**Comment by emag** [comment]
@gastaldi Thank you for your revewing!
I'd like to think about your comments, please hold this.

**Comment by emag** [comment]
Hmm, the failure is weird. I don't think it's related to my changes.

https://ci.wildfly-swarm.io/job/wildfly-swarm-pull-request-windows/org.wildfly.swarm$testsuite-management-console/195/testReport/junit/org.wildfly.swarm.management.console/ManagementConsoleTest/testHomePageShouldHaveCorrectTitle/

**Comment by kenfinnigan** [comment]
you're right @emag it's not.

Do you still want this held or we can merge?

**Comment by emag** [comment]
@kenfinnigan Thank you! I got it. Please merge this PR.

**Pull request merged**